src/test/java/org/folio/rest/api/PatronActionSessionAPITest.java

```diff
@@ -179,12 +185,99 @@ public void cannotUpdateNonExistentActionSession() throws InterruptedException,
+    createEntity(secondPatronId, "Check-out", DateTime.now().minus(1));
+    createEntity(secondPatronId, "Check-out", DateTime.now().minus(1));
+
+    assertThat(getExpiredSessionPatronIds("Check-in", 10, DateTime.now().minusDays(2)).size(),
+      is(1));
+    assertThat(getExpiredSessionPatronIds("Check-out", 10, DateTime.now()).size(),
+      is(2));
+    assertThat(getExpiredSessionPatronIds("Check-in", 10, DateTime.now().minusDays(2)).get(0),
+      is(firstPatronId));
+    assertThat(getExpiredSessionPatronIds("Check-out", 10, DateTime.now().minusDays(2)).get(0),
+      is(firstPatronId));
+    assertThat(getExpiredSessionPatronIds("Check-out", 2, DateTime.now()).size(),
+      is(2));
+  }
+
+  @Test
+  public void cannotGetPatronActionSessionStorageExpiredSessionPatronIds()
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+
+    checkGetExpiredSessionPatronIdsWithInvalidDate("Check-out", 10);
+  }
+
+  private JsonObject createSession(String actionType) {
     JsonObject jsonObject = new JsonObject()
       .put("id", UUID.randomUUID().toString())
       .put("patronId", UUID.randomUUID().toString())
       .put("loanId", existingLoanId)
       .put("actionType", actionType);
     return jsonObject;
   }
+
+  private JsonObject createEntity(String patronId, String actionType, DateTime createdDate) {
+
+    String userId = UUID.randomUUID().toString();
+    JsonObject metaData = new JsonObject()
+      .put("createdDate", createdDate.toString(ISODateTimeFormat.dateTime()))
+      .put("createdByUserId", userId)
+      .put("updatedDate", createdDate.toString(ISODateTimeFormat.dateTime()))
+      .put("updatedByUserId", userId);
+
+    JsonObject session = createSession(actionType);
+    session
+      .put("metadata", metaData)
+      .put("id", UUID.randomUUID().toString())
+      .put("patronId", patronId)
+      .put("loanId", existingLoanId);
+
+    CompletableFuture<String> future = new CompletableFuture<>();
+    PostgresClient
+      .getInstance(StorageTestSuite.getVertx(), TENANT_ID)
+      .save(PATRON_ACTION_SESSION, session, update -> future.complete(update.result()));
+    future.join();
+    return session;
+  }
+
+  private List<String> getExpiredSessionPatronIds(String actionType, int limit, DateTime lastActionDateLimit)
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+
+    URL url = StorageTestSuite.storageUrl("/patron-action-session-storage/expired-session-patron-ids",
+      "action_type", actionType, "last_time_action_limit",
+      lastActionDateLimit.toString(ISODateTimeFormat.dateTime()), "limit", Integer.toString(limit));
+
+    JsonResponse response = performRequest(url);
+    assertThat(response, isOk());
+    return response.getJson().getJsonArray("patronIds").getList();
+  }
+
+  private void checkGetExpiredSessionPatronIdsWithInvalidDate(String actionType, int limit)
```

Comment:
This appears to be only used once, why has it been extracted separate from that test?

Reply:
updated